gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GcsioTrackingHttpRequestInitializer.java

```diff
@@ -0,0 +1,157 @@
+import static com.google.cloud.hadoop.gcsio.GoogleCloudStorageStatistics.ACTION_HTTP_HEAD_REQUEST_FAILURES;
+import static com.google.cloud.hadoop.gcsio.GoogleCloudStorageStatistics.ACTION_HTTP_PATCH_REQUEST;
+import static com.google.cloud.hadoop.gcsio.GoogleCloudStorageStatistics.ACTION_HTTP_PATCH_REQUEST_FAILURES;
+import static com.google.cloud.hadoop.gcsio.GoogleCloudStorageStatistics.ACTION_HTTP_POST_REQUEST;
+import static com.google.cloud.hadoop.gcsio.GoogleCloudStorageStatistics.ACTION_HTTP_POST_REQUEST_FAILURES;
+import static com.google.cloud.hadoop.gcsio.GoogleCloudStorageStatistics.ACTION_HTTP_PUT_REQUEST;
+import static com.google.cloud.hadoop.gcsio.GoogleCloudStorageStatistics.ACTION_HTTP_PUT_REQUEST_FAILURES;
+
+import com.google.api.client.http.HttpExecuteInterceptor;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpRequestInitializer;
+import com.google.api.client.http.HttpResponse;
+import com.google.api.client.http.HttpResponseInterceptor;
+import com.google.common.flogger.GoogleLogger;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicLong;
+
+/** To track and update the statistics related to http requests */
+class GcsioTrackingHttpRequestInitializer implements HttpRequestInitializer {
+
+  private final HttpRequestInitializer delegate;
+
+  // To track the http statistics
+  private HashMap<GoogleCloudStorageStatistics, AtomicLong> httpStatistics;
+
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+
+  public GcsioTrackingHttpRequestInitializer(
+      HttpRequestInitializer delegate,
+      HashMap<GoogleCloudStorageStatistics, AtomicLong> httpStatistics) {
+    this.delegate = delegate;
+    this.httpStatistics = httpStatistics;
+  }
+
+  @Override
+  public void initialize(HttpRequest request) throws IOException {
+    if (delegate != null) {
+      delegate.initialize(request);
+    }
+    HttpExecuteInterceptor executeInterceptor = request.getInterceptor();
+    request.setInterceptor(
+        r -> {
+          if (executeInterceptor != null) {
+            executeInterceptor.intercept(r);
+          }
+          sethttprequestStats(r);
+        });
+    request.setResponseInterceptor(
+        new HttpResponseInterceptor() {
+          @Override
+          public void interceptResponse(HttpResponse httpResponse) throws IOException {
+            setHttpRequestFailureStats(httpResponse);
+          }
+        });
+  }
+
+  /**
+   * set the stats for the http requests
+   *
+   * @param r - request
+   */
+  private void sethttprequestStats(HttpRequest r) {
+
+    if (r.getRequestMethod() == "GET") {
+      httpStatistics.putIfAbsent(
+          GoogleCloudStorageStatistics.ACTION_HTTP_GET_REQUEST, new AtomicLong(0));
+      httpStatistics.get(ACTION_HTTP_GET_REQUEST).incrementAndGet();
+      logger.atInfo().log("GET Request - " + Objects.toString(r));
```

Comment:
Do we need logs ?  the code within the conditions is common, please simplify within a method